sdk/storage/Azure.Storage.Blobs.ChangeFeed/src/ChangeFeedFactory.cs

```diff
@@ -0,0 +1,191 @@
+            }
+            // Round start and end time if we are not using the cursor.
+            else
+            {
+                startTime = startTime.RoundDownToNearestHour();
+                endTime = endTime.RoundUpToNearestHour();
+            }
+
+            // Check if Change Feed has been abled for this account.
+            bool changeFeedContainerExists;
+
+            if (async)
+            {
+                changeFeedContainerExists = await containerClient.ExistsAsync().ConfigureAwait(false);
+            }
+            else
+            {
+                changeFeedContainerExists = containerClient.Exists();
+            }
+
+            if (!changeFeedContainerExists)
+            {
+                throw new ArgumentException("Change Feed hasn't been enabled on this account, or is currently being enabled.");
+            }
+
+            // Get last consumable
+            BlobClient blobClient = containerClient.GetBlobClient(Constants.ChangeFeed.MetaSegmentsPath);
+            BlobDownloadInfo blobDownloadInfo;
+            if (async)
+            {
+                blobDownloadInfo = await blobClient.DownloadAsync().ConfigureAwait(false);
+            }
+            else
+            {
+                blobDownloadInfo = blobClient.Download();
+            }
+
+            JsonDocument jsonMetaSegment;
+            if (async)
+            {
+                jsonMetaSegment = await JsonDocument.ParseAsync(blobDownloadInfo.Content).ConfigureAwait(false);
+            }
+            else
+            {
+                jsonMetaSegment = JsonDocument.Parse(blobDownloadInfo.Content);
+            }
+
+            lastConsumable = jsonMetaSegment.RootElement.GetProperty("lastConsumable").GetDateTimeOffset();
+
+            // Get year paths
+            years = await GetYearPaths(async, containerClient).ConfigureAwait(false);
+
+            // Dequeue any years that occur before start time
+            if (startTime.HasValue)
+            {
+                while (years.Count > 0
+                    && years.Peek().ToDateTimeOffset() < startTime.RoundDownToNearestYear())
+                {
+                    years.Dequeue();
+                }
+            }
+
+            if (years.Count == 0)
+            {
+                return new ChangeFeed();
+            }
+
+            string firstYearPath = years.Dequeue();
+
+            // Get Segments for first year
+            segments = await BlobChangeFeedExtensions.GetSegmentsInYear(
+                async: async,
+                containerClient: containerClient,
+                yearPath: firstYearPath,
+                startTime: startTime,
+                endTime: BlobChangeFeedExtensions.MinDateTime(lastConsumable, endTime))
+                .ConfigureAwait(false);
+
+            Segment currentSegment = await _segmentFactory.BuildSegment(
+                async,
+                containerClient,
+                segments.Dequeue(),
```

Comment:
The `segments[]` could be empty if there is no segment in that year after the `startTime`. I added a while loop to fetch the segments till it's not empty.

Reply:
fixed.